Enable Client PasswordHash

robertlestak opened the pull request:
The PRTG API enables the user to pass either a plain-text password, or a
password hash generated by the API. The NewClient function has been
modified to enable the user to pass either a password or a password hash
to the API wrapper.

Commit:
Enable Client PasswordHash

The PRTG API enables the user to pass either a plain-text password, or a
password hash generated by the API. The NewClient function has been
modified to enable the user to pass either a password or a password hash
to the API wrapper.

haidlir commented:
Hi @robertlestak,
Thanks for your PR. I have two suggestions for you.
1. To make sure that it's compatible to the existing/previous implementation of this package, could you add new function of NewClient (ie: NewClientWithHashedPass) instead of replacing existing function before merging this PR. Don't forget to create the test case.
2. Please add new example for the implementation of the new function.
Anything else looks good for me.
Thanks.

Commit:
Created new 'NewClientWithHashedPass' function

The previous commit modified the existing 'NewClient' function, which
would cause backwards-compatibility issues. The password hash
modification has been moved to a separate function to ensure it does not
break existing implementations of the NewClient function.

haidlir commented:
Hi @robertlestak,
Thanks for your update.
1. I think that you forget to create the test for prtg.api.go line 102. For this, you can copy TestGetPrtgVersion, start from this and alter the test function to use the NewClientWithHashedPass function.
<img width="835" alt="screen shot 2018-07-24 at 12 54 11" src="https://user-images.githubusercontent.com/8347608/43119466-b82acad4-8f40-11e8-93a7-6600ee7a4d68.png">

2. (it's optional), but I realized that it might be good if we also provide function to request hashed password from the PRTG Server that later will be used as the NewClientWithHashedPass argument.
Thanks.

Commit:
Increased test coverage for NewClientWithHashedPass

robertlestak commented:
@haidlir I just updated the tests and prtg_api.go now has 100% coverage.

I didn't have the time this week to implement the password hash functionality, however all it takes is a call to `/api/getpasshash.htm` with the Basic Auth username / password combo and it will return a string which is your hashed pass, so implementing it shouldn't take much time.

haidlir commented:
@robertlestak, your PR is already merged.. Thanks a lot for your contribution.
Keep up the good work.
